### ISSUE TITLE
Some algorithmic problems

### ISSUE THREAD

**Issue by sunkun1997** [microsoft/LightGBM#5449]
Sorry to bother you again, but I still have some question  to solve.

I've read the following code and come to some conclusions
https://github.com/microsoft/LightGBM/blob/be7f321382b22a1ab536965c5fdfefe131e488f6/src/treelearner/feature_histogram.hpp#L880

After find the best split threshold, the left_sum_gradient is calculated by   
$$G_{L}=\sum_{i=0}^mg_i$$,  
the left_sum_hessian is calculated by  
$$H_{L}=\sum_{i=0}^mh_i$$
the right_sum_gradient is calculated by  
$$G_{R}=\sum_{j=0}^ng_j$$
the right_sum_hessian is calculated by
$$H_{R}=\sum_{j=0}^nh_j$$

so that the left_output and right_output are calculated by  
$$O_L = \frac{-G_L} {H_L + \lambda_{l2}}$$
$$O_R = \frac{-G_R} {H_R + \lambda_{l2}}$$
and parent_output are calculated by  
$$O_P = \frac{-(G_L+G_R)} {(H_L+H_R) + \lambda_{l2}}$$

so the split_gain is calculated by  
$$- (2G_LO_L + H_LO_L^2) + (- (2G_RO_R+ H_RO_R^2)) - (- (2(G_L+G_R)O_P + (H_L+H_R)O_P ^2))$$

My question are:

1. How to calculate value? Value is the predicted value of the model. I can call model.trees_to_dataframe() to get it by Python.
2. How to update score? Because I need it to calculate the gradient so I want to know how is it updated.
3. What is the connection between score and value?

**Comment by sunkun1997** [comment]
I think I tried it out. value is calculated by output*learning_rate, and score is calculated by sum of output. is that right?
